transport/src/main/java/io/netty/channel/oio/OioByteStreamChannel.java

```diff
@@ -126,25 +126,19 @@ protected void doWriteFileRegion(FileRegion region) throws Exception {
             outChannel = Channels.newChannel(os);
         }
 
-        long written = 0;
-        for (;;) {
-            long localWritten = region.transferTo(outChannel, written);
-            if (localWritten == -1) {
-                checkEOF(region);
+        long bytesToTransfer = obj.readableBytes();
+        long totalTransferred = 0;
+        while (obj.isReadable()) {
+            long transferred = obj.readTo(outChannel, obj.readableBytes());
+            if (transferred == -1) {
```

Comment:
@normanmaurer AFAICT, this method should never return `-1`.  Was this a mistake? Should we be checking for `0` instead?